src/metabase/sync/analyze/fingerprint.clj

```diff
@@ -47,23 +49,25 @@
   [table :- i/TableInstance, fields :- [i/FieldInstance]]
   (let [rff (fn [_metadata]
               (redux/post-complete
-                (fingerprinters/fingerprint-fields fields)
-                (fn [fingerprints]
-                  (reduce (fn [count-info [field fingerprint]]
-                            (cond
-                              (instance? Throwable fingerprint)
-                              (update count-info :failed-fingerprints inc)
-
-                              (some-> fingerprint :global :distinct-count zero?)
-                              (update count-info :no-data-fingerprints inc)
-
-                              :else
-                              (do
-                                (save-fingerprint! field fingerprint)
-                                (update count-info :updated-fingerprints inc))))
-                          (empty-stats-map (count fingerprints))
-                          (map vector fields fingerprints)))))]
-    (metadata-queries/table-rows-sample table fields rff {:truncation-size truncation-size})))
+               (fingerprinters/fingerprint-fields fields)
+               (fn [fingerprints]
+                 (reduce (fn [count-info [field fingerprint]]
+                           (cond
+                             (instance? Throwable fingerprint)
+                             (update count-info :failed-fingerprints inc)
+
+                             (some-> fingerprint :global :distinct-count zero?)
+                             (update count-info :no-data-fingerprints inc)
+
+                             :else
+                             (do
+                               (save-fingerprint! field fingerprint)
+                               (update count-info :updated-fingerprints inc))))
+                         (empty-stats-map (count fingerprints))
+                         (map vector fields fingerprints)))))
+        driver (driver.u/database->driver (table/database table))
+        opts {:truncation-size truncation-size}]
+    (driver.workarounds/table-rows-sample driver table fields rff opts)))
```

Comment:
It reads a little weird to me to have `driver.workarounds/table-rows-sample` be a normal method with a `:default` implementation that you'd normally use for day-to-day rows sampling. It's also nice too if the driver methods are mostly in one place -- I think it was a mistake to split up the SQL and JDBC driver methods over a bunch of namespaces since it makes it hard to figure out what methods are available to implement. 

Would it make more sense just to have this live in `metabase.driver` with all of the other non-SQL/non-JDBC driver multimethods? A lot of the methods in there are already ones you only need to implement in certain circumstances

Just my two cents 🤷 
I don't feel that strongly about this TBH

Reply:
Oh yeah, now that you mention that, that is pretty much what I said back then. I don't think this needs to be an "experimental do not implement" method anymore however. I think the main reasoning we were discussing that is that we weren't sure this new method was a good idea at the time and we didn't want a lot of people going around implementing it if we decided to remove it in the future. But I think at this point it's pretty clear that we do want to leave it in going forward.